k8sutils/pkg/workload/workloadkinds.go

```diff
@@ -0,0 +1,108 @@
+package workload
+
+import (
+	"errors"
+	"strings"
+
+	v1 "k8s.io/api/apps/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// This go file contains utils to handle the kind of odigos workloads.
+// it allows transforming deployments / daemonsets / statefulsets from one representation to another
+
+// 1. the pascal case representation of the workload kind
+// it is used in k8s api objects as the `Kind` field.
+type WorkloadKind string
+
+const (
+	WorkloadKindDeployment  WorkloadKind = "Deployment"
+	WorkloadKindDaemonSet   WorkloadKind = "DaemonSet"
+	WorkloadKindStatefulSet WorkloadKind = "StatefulSet"
+)
+
+// 2. the lower case representation of the workload kind
+// is used in odigos with the object name for instrumentation config and runtime details
+type WorkloadKindLowerCase string
+
+const (
+	WorkloadKindLowerCaseDeployment  WorkloadKindLowerCase = "deployment"
+	WorkloadKindLowerCaseDaemonSet   WorkloadKindLowerCase = "daemonset"
+	WorkloadKindLowerCaseStatefulSet WorkloadKindLowerCase = "statefulset"
+)
+
+var ErrKindNotSupported = errors.New("workload kind not supported")
+
+func IsErrorKindNotSupported(err error) bool {
+	return err == ErrKindNotSupported
+}
+
+func IgnoreErrorKindNotSupported(err error) error {
+	if IsErrorKindNotSupported(err) {
+		return nil
+	}
+	return err
+}
+
+func WorkloadKindLowerCaseFromKind(pascalCase WorkloadKind) WorkloadKindLowerCase {
```

Comment:
It is equivalent to `strings.ToLower`. I'm not sure why we need types for both pascal and lowercase. We can have just the pascal one and use `ToLower` when needed in the utils functions

Reply:
yes, since it's used in several places, I choose to have a type for it as enum